File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/repo/model/SimpleMarker.java
package com.example.ama.android2_lesson03.repo.model;

import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.Marker;

public class SimpleMarker {
    private LatLng position;
    private String title;

    public LatLng getPosition() {
        return position;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public SimpleMarker(String title, LatLng position) {
        this.title = title;
        this.position = position;
    }

    public static SimpleMarker getFromMarker(Marker marker) {
        return new SimpleMarker(marker.getTitle(), marker.getPosition());
    }

    @Override
    public String toString() {
        return title;
    }
}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/repo/model/SimpleMarker.kt
package com.example.ama.android2_lesson03.repo.model

import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.Marker

/**
 * Model with basic marker parameters for saving to favorites
 */
data class SimpleMarker(val address: String, val position: LatLng) {
    var title: String = address

    companion object {
        fun getFromMarker(marker: Marker): SimpleMarker = SimpleMarker(marker.snippet, marker.position)
        fun simpleMarkerWithTitle(title: String, address: String, position: LatLng): SimpleMarker {
            val marker = SimpleMarker(address, position)
            marker.title = title
            return marker
        }
    }

    override fun toString() = title
}
